### PR TITLE
feat: link wasm-keys

### LINKED ISSUE #35: add hotkey validation
upon declaring a hotkey, we should check if the keybinding is valid.

We can do this since `wasm_bindgen` uses the `KeyboardEvent`'s `.key()`.

MDN: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values

### PR DESCRIPTION
**Pull request by friendlymatthew**
This PR should be the last feature for our alpha release. [wasm-keys](https://github.com/gaucho-labs/wasm-keys) is a library that strongly types the `KeyboardEvent's` `key()` values. 

In addition, hotkey logic resides there.

closes #35

### CODE REVIEW

File: leptos_hotkeys/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ leptos = "0.6"
 log = { version = "0.4", optional = true }
 wasm-bindgen = "0.2"
 web-sys = "0.3"
+wasm-keys = { git = "https://github.com/gaucho-labs/wasm-keys" }
```
Comment (mondeja) [inline leptos_hotkeys/Cargo.toml:17]: This is really unsafe. Would you mind to release a version of wasm-keys?

File: leptos_hotkeys/src/use_hotkeys.rs [hunk 2]
```diff
@@ -109,11 +95,13 @@ pub fn use_hotkeys_ref_scoped<T>(
                         .find(|hotkey| is_hotkey_match(hotkey, &mut pressed_keys))
                     {
                         if cfg!(feature = "debug") {
-                            let message = format!("%cfiring hotkey: {}", &matching_hotkey);
+                        /*    let message = format!("%cfiring hotkey: {}", &matching_hotkey);
```
Comment (mondeja) [inline leptos_hotkeys/src/use_hotkeys.rs:98]: Shouldn't be a `to_string()` here?

File: leptos_hotkeys/src/use_hotkeys.rs [hunk 3]
```diff
@@ -67,11 +53,11 @@ pub fn use_hotkeys_scoped(
                     .find(|hotkey| is_hotkey_match(hotkey, &mut pressed_keyset))
                 {
                     if cfg!(feature = "debug") {
-                        let message = format!("%cfiring hotkey: {}", &matching_hotkey);
+                        /*let message = format!("%cfiring hotkey: {}", &matching_hotkey);
```
Comment (mondeja) [inline leptos_hotkeys/src/use_hotkeys.rs:56]: Here also.

**Review by mondeja — CHANGES_REQUESTED** [review]
Additionally, you can fix the format running `cargo make format`.

File: leptos_hotkeys/Cargo.toml (hunk 1 above)
Comment (friendlymatthew) [reply leptos_hotkeys/Cargo.toml:17]: Yes -- sorry I was going to do this during the weekend

**Comment by friendlymatthew** [comment]
@mondeja I'm starting to think we shouldn't merge this.

The design is sound, but I'm not sure how useful `wasm-keys` will be as it's own library. Plus, strongly typing the keys don't really do anything.

But I'm happy to hear your thoughts and if you feel like this is worth trying, I'm happy to continue to work on this

**Comment by mondeja** [comment]
I haven't tried it but if you say it's not worth it, I believe you. In any case, we can always revisit the idea later again if it adds value.

**Pull request closed without merging**